### PR TITLE
Unix build instructions

### PR DESCRIPTION
**Pull request by doc-hex**
Thanks for contributing to COLDCARD source code, please be as descriptive as possible.

- If you haven't yet, please check out our docs https://coldcard.com/docs/
- Have you tested your work yet? You can do it with the simulator https://github.com/Coldcard/firmware/blob/master/unix/simulator.py

*By submitting this work you agree to the [Individual Contributor License Agreement (CLA)](https://raw.githubusercontent.com/Coldcard/firmware/master/CONTRIBUTING.md)*

---

### CODE REVIEW

**Comment by doc-hex** [comment]
I would rather these patch files get upstreamed into `libngu`

**Comment by doc-hex** [comment]
- broader problem: unix changes need to work on both MacOS and Linux ... you've selected the Linux options, but what was there is MacOS (unix) compatible
- both arc4random and address decoding things

**Comment by scgbckbone** [comment]
yeah, consider this more to be POC, I created PR prematurely. Now I'm having a lot of issues running tests against simulator (`pytest --sim`) - lot of them failing

1. I need to make this work with linux (all test should pass - > am I correct expecting that all should pass via `pytest --sim`?)

**Comment by dipunm** [comment]
Hey @doc-hex thanks for making this PR, I managed to get the simulator running!

FYI: getting my head around submodules seems to be impossible so this is the set of instructions I had to follow to get this working (when merged to master, I imagine this will be easier):

```bash
git clone git@github.com:Coldcard/firmware
cd firmware
git remote add scgbckbone git@github.com:scgbckbone/firmware
git fetch --all
git checkout unix_build_instructions
git submodule update --init
```

After that, I was able to run the steps from applying `unix_addr.patch` all the way to `./simulator.py`

Without exactly the above steps, I kept getting compiler errors.

**Comment by scgbckbone** [comment]
one thing to check here is ngu submodule commit - now it tracks master (there are few more changes than just linux compatibility)

**Comment by doc-hex** [comment]
- external/libngu hash has changed on master, and this PR should update (rebase?) to that

**Comment by doc-hex** [comment]
@scgbckbone will look at this next week

File: unix/variant/ffilib.py [hunk 1]
```diff
@@ -14,7 +14,8 @@ def open(name, maxver=10, extra=()):
     except KeyError:
         pass
     def libs():
-        if sys.platform == "linux":
+        # TODO what about sys.platform == "darwin" ??
+        if sys.platform in ["linux", "linux2", "coldcard-unix"]:
```
[the remark this answers is not in the document]
Comment (scgbckbone) [reply unix/variant/ffilib.py:18]: if `sys.platform` == "darwin" then we get to `else` part where dylib is properly handled - or am I missing something ?

**Comment by scgbckbone** [comment]
closing this - this https://github.com/Coldcard/firmware/pull/104 should go in and contains all changes from this PR

File: unix/variant/ffilib.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (doc-hex) [reply unix/variant/ffilib.py:18]: Fixed better in 6b02f49 ... because Darwin (MacOS) broke, and fact is `sys.platform` is wrong/misleading here.

**Pull request closed without merging**